compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -806,18 +831,30 @@ public BLangNode transform(ConstantDeclarationNode constantDeclarationNode) {
 
         // Check whether the value is a literal. If it is not a literal, it is an invalid case. So we don't need to
         // consider it.
+        // Note - This logic has been special cased for unary expressions as `+` and `-` numeric literals no longer
+        // go on the literal path but as a unary expression.
```

Comment:
For constants without a type node like : `const a = -1; ` in master, it is allowed only for literals but since `-1` is a unary expression, In all the places where it was only allowed for literals I updated it to also allow unary expressions with `+` or `-` numeric literals.